### ISSUE TITLE
HELICS 2.0 development

### ISSUE THREAD

**Issue by phlptp** [GMLC-TDC/HELICS-src#324]
@bpalmintier @kdheepak @jcfuller1 @nightlark @HimanshuJain17 @shwethanidd @afisher1 
And any others who want to comment

I am considering starting a 2.0 branch for development of breaking changes.  The concept is this branch would be developed in parallel with the exising 1.X branches.  Any non-breaking changes could go first to develop and then merged into this branch,  But just looking at a few things there are some changes that need time to "bake" and get tested so this branch would allow us to "tinker" a little.  Also looking at some major changes to the Language interface API's that will solve some issues but would be major breaking changes as well. 

One reason for this now is that I started working on one change but realized it would technically be a breaking change in the Core API,  the only user that I know of is the Application API, but it is still a breaking change and would require an update in major version number if merged immediately.  

Just wanted others thoughts on this development model in working towards a 2.0 release.

**Comment by jcfuller1** [comment]
@phlptp - This works for me. It's close enough to a GitFlow work flow that at least it won't confuse me!

**Comment by phlptp** [comment]
We have a HELICS_2_0 branch now. This will be the development branch for anything going into HELICS 2.0 so it won't be stable for a while, and will include some significant API changes.  Though for the moment the changes are not significant on the API side.

**Comment by phlptp** [comment]
I will keep tinkering with this branch and we can add in any API updates as we go but I will close the issue for now
